build/configs/common_download.sh

```diff
@@ -0,0 +1,390 @@
+				continue
+			elif [[ "${parts[$partidx]}" == "ss" ]];then
+				continue
+			else
+				echo ""
+				echo "=========================="
+				echo "Erasing ${parts[$partidx]} partition"
+				echo "=========================="
+			fi
+
+			board_erase $TTYDEV ${offsets[$partidx]} ${sizes[partidx]}
+		done
+	else
+		found_kernel=false
+		for partidx in ${!parts[@]}; do
+			if [[ $2 == "kernel" || $2 == "KERNEL" ]];then
+				if [[ "${parts[$partidx]}" != "kernel" ]];then
+					continue
+				elif [[ $found_kernel == false ]];then
+					found_kernel=true
+				else
+					continue
+				fi
+			elif [[ $2 == "ota" || $2 == "OTA" ]];then
+				if [[ "${parts[$partidx]}" != "kernel" ]];then
+					continue
+				elif [[ $found_kernel == false ]];then
+					found_kernel=true
+					continue
+				else
+					flash_ota=false
+				fi
+			elif [[ $2 == "ss" || $2 == "SS" ]];then
+				if [[ "${parts[$partidx]}" != "ss" ]];then
+					continue
+				fi
+			elif [[ $2 == "userfs" || $2 == "USERFS" ]];then
+				if [[ "${parts[$partidx]}" != "userfs" ]];then
+					continue
+				fi
+			else
+				printf "\n## Invalid Option ##\n\n"
+				dwld_help
+				break
+			fi
+
+			echo ""
+			echo "=========================="
+			echo "Erasing ${parts[$partidx]} partition"
+			echo "=========================="
+			board_erase $TTYDEV ${offsets[$partidx]} ${sizes[partidx]}
+		done
+	fi
+	echo ""
+	echo "Erase COMPLETE!"
+}
+
+# Start here
+
+echo -e $WARNING
+
+cmd_args=$(echo $@ | tr '[:upper:]' '[:lower:]')
+
+# Treat adding the USB rule first
+usb_rule ${cmd_args[@]};
+
+sanity_check;
+
+parts=$(get_configured_partitions)
+IFS=',' read -ra parts <<< "$parts"
+
+sizes=$(get_partition_sizes)
+IFS=',' read -ra sizes <<< "$sizes"
+
+#Calculate Flash Offset
+num=${#sizes[@]}
+offsets[0]=`printf "0x%X" ${FLASH_START_ADDR}`
+
+for (( i=1; i<=$num-1; i++ ))
+do
+	val=$((sizes[$i-1] * 1024))
+	offsets[$i]=`printf "0x%X" $((offsets[$i-1] + ${val}))`
+done
+
+#Dump Info
+echo ""
+echo "================================ < Flash Partition Information > ================================"
+printf "\rNAME       :"
+printf '  %12s' "${parts[@]}"
+printf "\r\nSIZE(in KB):"
+printf '  %12s' "${sizes[@]}"
+printf "\r\nAddr       :"
+printf '  %12s' "${offsets[@]}"
+printf "\n"
+echo "================================================================================================="
+echo ""
+
+if test $# -eq 0; then
+	echo "FAIL!! NO Given partition. Refer \"PARTITION NAMES\" above."
+	dwld_help 1>&2
+	exit 1
+fi
+
+uniq_parts=($(printf "%s\n" "${parts[@]}" | sort -u));
+
+#Validate arguments
+for i in ${cmd_args[@]};do
+
+	if [[ "${i}" == "ALL" || "${i}" == "all" || "${i}" == "erase" || "${i}" == "ERASE" ]];then
+		continue;
+	fi
+
+	for j in ${uniq_parts[@]};do
+		if [[ "${i}" == "${j}" ]];then
+			result=yes
+		fi
+	done
+
+	if [[ "$result" != "yes" ]];then
+		echo "FAIL!! Given \"${i}\" partition is not available. Refer \"PARTITION NAMES\" above."
+		dwld_help
+		exit 1
+	fi
+	result=no
```

Comment:
Let us move this to line no  355. I think it makes more sense to initialized the variable there.